### PR TITLE
GNUI-72-fix-tooltip

### PR DESCRIPTION
**Pull request by Horay**
When merged this PR will change the tooltip component

### CODE REVIEW

File: src/components/tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -19,205 +17,187 @@ const setColor = (color: string) => {
...
-
-  p {
-    margin:0;
-    display:inline-block;
+  &:after {
+    border-color: transparent;
+    border-top-color: ${theme.colors.primary};
+    border-width: 8px;
+    margin-left: -8px;
+
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:82]: unnecessary empty line

File: src/components/tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -1,12 +1,10 @@
-import React, { FunctionComponent, useState } from "react";
+import React, { FunctionComponent, useState, useEffect, useRef } from "react";
 import styled, { css } from "styled-components";
 import theme from "../../theme";
 
 interface TooltipProps {
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:5]: ITooltip sounds better

File: src/components/tooltip/Tooltip.tsx (hunk 1 above)
Comment (Horay) [reply src/components/tooltip/Tooltip.tsx:82]: ok

File: src/components/tooltip/Tooltip.tsx (hunk 2 above)
Comment (Horay) [reply src/components/tooltip/Tooltip.tsx:5]: ok

File: src/components/tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -19,205 +17,187 @@ const setColor = (color: string) => {
...
+  max-width: 256px;
+  font-size: ${theme.fontSizes.sm};
+  line-height: 1rem;
+  padding: ${theme.spacing.spacing02};
+  background-color: ${theme.colors.primary};
+  color: ${theme.colors.snowWhite};
+  border-radius: ${theme.radius.md};
+  z-index: ${theme.zindex.sticky};
+  animation-name: fadeInUp;
+  -webkit-animation-name: fadeInUp;
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:58]: css prefixes are not necessary

File: src/components/tooltip/Tooltip.tsx [hunk 4]
```diff
@@ -19,205 +17,187 @@ const setColor = (color: string) => {
...
-  }
-  &.top {
-    bottom: calc(100% + 8px);
-    left: -4px;
-  }
-  &.bottom {
-    top: calc(100% + 8px);
-    left: -4px;
-    right:0;
+  max-width: 256px;
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:49]: rem instead of px

File: src/components/tooltip/Tooltip.tsx [hunk 5]
```diff
@@ -19,205 +17,187 @@ const setColor = (color: string) => {
...
-            right: 12px;
-            left:auto;
-        `
-        default:
-          return css`
-            left: calc(50% - 7px);
-            right:auto;
-        `
-      }
+  @-webkit-keyframes fadeInUp {
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:109]: css prefixes are not necessary

File: src/components/tooltip/Tooltip.tsx [hunk 6]
```diff
@@ -19,205 +17,187 @@ const setColor = (color: string) => {
...
-    right:0;
+  max-width: 256px;
+  font-size: ${theme.fontSizes.sm};
+  line-height: 1rem;
+  padding: ${theme.spacing.spacing02};
+  background-color: ${theme.colors.primary};
+  color: ${theme.colors.snowWhite};
+  border-radius: ${theme.radius.md};
+  z-index: ${theme.zindex.sticky};
+  animation-name: fadeInUp;
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:57]: there is a shorthand for animation:

File: src/components/tooltip/Tooltip.tsx [hunk 7]
```diff
@@ -1,12 +1,10 @@
...
 interface TooltipProps {
-  name?: string;
-  caption?: string;
-  position?: "left" | "right" | "top" | "bottom";
-  arrowPosition?: "left" | "right";
+  caption: string;
+  position?: "left" | "right";
   status?: "danger" | "warning" | "success" | "notification";
   id?: string;
   className?: string;
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:10]: className ?

File: src/components/tooltip/Tooltip.tsx [hunk 8]
```diff
@@ -1,12 +1,10 @@
...
 
 interface TooltipProps {
-  name?: string;
-  caption?: string;
-  position?: "left" | "right" | "top" | "bottom";
-  arrowPosition?: "left" | "right";
+  caption: string;
+  position?: "left" | "right";
   status?: "danger" | "warning" | "success" | "notification";
   id?: string;
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:9]: "id"  is not used anywhere

File: src/components/tooltip/Tooltip.tsx [hunk 9]
```diff
@@ -19,205 +17,187 @@ const setColor = (color: string) => {
     : color;
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:17]: The setColor is common in application let's put it into utils, and import where needed.

File: src/components/tooltip/Tooltip.tsx [hunk 10]
```diff
@@ -1,15 +1,11 @@
...
 
-interface TooltipProps {
-  name?: string;
-  caption?: string;
-  position?: "left" | "right" | "top" | "bottom";
-  arrowPosition?: "left" | "right";
+interface ITooltip {
+  caption: string;
+  position?: "left" | "right";
   status?: "danger" | "warning" | "success" | "notification";
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:8]: Maybe let's use "color" similar like in Button component?

File: src/components/tooltip/Tooltip.tsx [hunk 11]
```diff
@@ -19,205 +15,169 @@ const setColor = (color: string) => {
...
+  max-width: 16rem;
+  font-size: ${theme.fontSizes.sm};
+  line-height: 1rem;
+  padding: ${theme.spacing.spacing02};
+  background-color: ${theme.colors.primary};
+  color: ${theme.colors.snowWhite};
+  border-radius: ${theme.radius.md};
+  z-index: ${theme.zindex.sticky};
+  opacity: 0;
+  animation: 0.2s ease-in both fadeInUp;
```
Comment (raczyk) [inline src/components/tooltip/Tooltip.tsx:56]: Let's import transition from theme ${theme.transition}.

**Review by janosio-nordcloud — APPROVED** [review]

**Review by janosio-nordcloud — APPROVED** [review]

**Review by raczyk — APPROVED** [review]

**Pull request merged**